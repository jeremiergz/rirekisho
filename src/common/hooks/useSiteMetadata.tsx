import { graphql, useStaticQuery } from 'gatsby';
import { IGatsbyImageData } from 'gatsby-plugin-image';

type SiteMetadataQueryResponse = {
  banner: IGatsbyImageData;
  site: {
    siteMetadata: {
      authorName: string;
      authorTwitterUsername: string;
      color: string;
      description: string;
      keywords: string[];
      license: string;
      name: string;
      repositoryURL: string;
      siteUrl: string;
      version: string;
    };
  };
};

function useSiteMetadata(): {
  banner: SiteMetadataQueryResponse['banner'];
} & SiteMetadataQueryResponse['site']['siteMetadata'] {
  const rawData = useStaticQuery<SiteMetadataQueryResponse>(graphql`
    query SiteMetadataQuery {
      banner: file(relativePath: { eq: "cover.png" }) {
        childImageSharp {
          gatsbyImageData(formats: [AUTO, WEBP, AVIF], height: 630, placeholder: BLURRED, quality: 100, width: 1200)
        }
      }
      site {
        siteMetadata {
          authorName
          authorTwitterUsername
          color
          description
          keywords
          license
          name
          repositoryURL
          siteUrl
          version
        }
      }
    }
  `);

  return { banner: rawData.banner, ...rawData.site.siteMetadata };
}

export type { SiteMetadataQueryResponse };
export default useSiteMetadata;
