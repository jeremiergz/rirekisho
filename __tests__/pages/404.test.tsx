import LayoutComponent from '@/common/components/layout/Layout';
import useSiteMetadata, * as Layout from '@/common/hooks/useSiteMetadata';
import NotFoundPage from '@/pages/404';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import mockData from './mock.json';

describe('pages/404 Test Suite', () => {
  beforeAll(() => {
    const { layoutData } = mockData;

    jest.spyOn(Layout, 'default').mockReturnValue(layoutData as unknown as ReturnType<typeof useSiteMetadata>);
  });

  it('matches snapshot', () => {
    const jsx = (
      <LayoutComponent>
        <NotFoundPage />
      </LayoutComponent>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
