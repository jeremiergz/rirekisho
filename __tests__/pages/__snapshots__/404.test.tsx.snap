// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/404 Test Suite matches snapshot 1`] = `
<div
  className="transition-opacity duration-200 ease-out opacity-0"
/>
`;
